tools/tool-testing/galaxy-utils.js

```diff
@@ -0,0 +1,251 @@
+  }
+  if (! GALAXY_USERNAME ||
+      ! GALAXY_PASSWORD) {
+    selftest.fail(
+      "Can't use test account with Galaxy. " +
+       "Please specify GALAXY_USERNAME and GALAXY_PASSWORD.\n");
+  }
+  if (! process.env.APP_MONGO) {
+    selftest.fail(
+      "Please provide an APP_MONGO url to use for deployed apps.\n");
+   }
+});
+
+// Login to Galaxy with environment-variable credentials passed in by the user.
+//
+// Unlike the normal `meteor deploy` Galaxy is not yet publically available, so
+// we don't want to use the publically-accessible test account here.
+exports.loginToGalaxy = selftest.markStack(function (sandbox) {
+  var user = GALAXY_USERNAME;
+  var pass = GALAXY_PASSWORD;
+  testUtils.login(sandbox, user, pass);
+});
+
+// Curl an app running on Galaxy. Automatically follow redirects.
+//
+// Dealing with DNS on Galaxy can be complicated. The standard way to ensure
+// that we hit the right app on the right Galaxy is to curl the galaxy origin,
+// with the host header set to our query app.
+exports.curlToGalaxy = selftest.markStack(function (url) {
+  var hostHeader = "host: " + url;
+  var galaxyOrigin = GALAXY_URL;
+  return runCurl("-vLH", hostHeader, galaxyOrigin);
+});
+
+// String we expect to hit on 200 OK.
+exports.httpOK = "HTTP/1.1 200 OK";
+
+// String we expect to hit when using SSL.
+exports.httpRedirect = "HTTP/1.1 307 Temporary Redirect";
+
+// We expect containers to take some time to startup.
+//
+// In the future, we can use this function to poll whether the containers have
+// started and maybe our tests will be faster.
+exports.waitForContainers = selftest.markStack(function () {
+  // We are not spinning up any containers in mock mode, so don't wait too long.
+  var waitTime = process.env.GALAXY_MOCK_MODE ?
+    1000 : 1000 * 10 * utils.timeoutScaleFactor;
+  utils.sleepMs(waitTime);
+});
+
+// Deploy an app against a Galaxy
+//
+// When we deploy to Galaxy, we need to specify a Mongo URL and wait a little
+// longer for the app to spin up.
+//
+// Options:
+//   - settings: app settings, NOT including mandatory galaxy settings
+//     such as MONGO_URL
+//   - appName: app name to use; will be generated randomly if not
+//     provided
+//   - templateApp: the name of the template app to use. defaults to
+//    'simple-app'
+//   - useOldSettings: don't make a new settings object this app! This is a
+//     redeploy, so reuse the settings that Galaxy has saved.
+//
+exports.createAndDeployApp =  selftest.markStack(function (sandbox, options) {
+  options = options || {};
+  var settings = options.settings;
+  var appName = options.appName || testUtils.randomAppName();
+
+  // The simple app contains standart app packages and some small bits of code
+  // so that we can check that it is being served correctly. Let's use that as
+  // our default.
+  var templateApp = options.templateApp || 'simple-app';
+
+  // Create the new galaxy settings.
+  var galaxySettings = {};
+  galaxySettings[appName] = {
```

Comment:
Hah, that's what I get for using `galaxy.meteor.com` as the app to copy settings from. Not very smart of me. Somehow, though, this actually worked, I swear (probably because none of my apps actually used the mongo db). Will fix.